src/main/java/org/folio/inventory/resources/Instances.java

```diff
@@ -359,29 +393,82 @@ private void makeInstanceResponse(
     CollectionResourceClient relatedInstancesClient = createInstanceRelationshipsClient(routingContext, context);
 
     if (relatedInstancesClient != null) {
-      relatedInstancesClient.getMany(query, (Response result) -> {
-        List<InstanceRelationshipToParent> parentInstanceList = new ArrayList();
-        List<InstanceRelationshipToChild> childInstanceList = new ArrayList();
-        if (result.getStatusCode() == 200) {
-          JsonObject json = result.getJson();
-          List<JsonObject> relationsList = JsonArrayHelper.toList(json.getJsonArray("instanceRelationships"));
-          relationsList.forEach(rel -> {
-            if (rel.getString(InstanceRelationship.SUPER_INSTANCE_ID_KEY).equals(instance.getId())) {
-              childInstanceList.add(new InstanceRelationshipToChild(rel));
-            } else if (rel.getString(InstanceRelationship.SUB_INSTANCE_ID_KEY).equals(instance.getId())) {
-              parentInstanceList.add(new InstanceRelationshipToParent(rel));
-            }
-          });
+      CompletableFuture<Response> relatedInstancesFetched = new CompletableFuture<>();
+
+      relatedInstancesClient.getMany(query, relatedInstancesFetched::complete);
+
+      return relatedInstancesFetched
+        .thenCompose(response -> withInstanceRelationships(instance, response));
+    }
+    return completedFuture(null);
+  }
+
+  private void successResponse(RoutingContext routingContext, WebContext context,
+                               Instance instance) {
+
+    JsonResponse.success(routingContext.response(), toRepresentation(instance,
+      new ArrayList<>(), new ArrayList<>(),
+      instance.getPrecedingTitles(), instance.getSucceedingTitles(), context));
+  }
+
+  private CompletableFuture<InstancesResponseWrapper> withInstancesRelationships(InstancesResponseWrapper instancesResponseWrapper,
+                                                                                 Response result) {
```

Comment:
Please can this be formatted with standard indentation

Reply:
Personally I would include both parameters on the second line, as the first line is already quite long